instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,311 @@
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testOneBillingSessionSentForPausedAndResumedFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionsForMixedFreeDriveAndActiveGuidanceTrip() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(3, billingRequests.size)
+        assertTrue(billingRequests[0].isFreeDriveSession)
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests[0].duration)
+        assertTrue(billingRequests[1].isActiveGuidanceSession)
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests[1].duration)
+        assertTrue(billingRequests[2].isFreeDriveSession)
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests[2].duration)
+    }
+
+    @Test
+    fun testBillingSessionsForStoppedAndResumedActiveGuidanceTrip() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
```

Comment:
Wait, you never stopped the session.

Reply:
It's not required as `clearNavigationRoutesAndWaitForUpdate()` begins free drive session and triggers active guidance billing session being sent. But I can add `stopTripSession()` it doesn't change anything.